src/DocBlock/TagComparator.php

```diff
@@ -19,23 +19,62 @@
  * together, or kept apart.
  *
  * @author Graham Campbell <hello@gjcampbell.co.uk>
+ * @author Jakub Kwaśniewski <jakub@zero-85.pl>
  */
 final class TagComparator
 {
     /**
      * Groups of tags that should be allowed to immediately follow each other.
+     *
+     * @internal
      */
-    private static array $groups = [
+    public const TAG_GROUPS = [
         ['deprecated', 'link', 'see', 'since'],
         ['author', 'copyright', 'license'],
         ['category', 'package', 'subpackage'],
         ['property', 'property-read', 'property-write'],
     ];
 
+    /**
+     * @var string[][]
+     */
+    private array $groups = [];
+
+    private function __construct()
+    {
+    }
```

Comment:
This class is not internal, and there are a few BC breaks. I don't think anyone would be affected by these though...

Reply:
I've checked this class before making changes. It has no clients except PhpdocSeparationFixer. @jrmajor, should I leave it as it is?

Reply:
It can be used by other packages, like https://github.com/kubawerlos/php-cs-fixer-custom-fixers or https://github.com/laravel/pint.

To avoid BC break you can revert changes in this class, add `@deprecated` to it, then
a) inline the new logic into the fixer, as it's unlikely it would be useful anywhere else, and it's not that much code,
b) move the new version to a new `@internal` class, as it's current name is strange anyway,

...or you can leave it as is and hope that mainatiners will accept it, as it's unlikely to break anyone's code anyway :) In this case, you may also add `@internal` to it to avoid this problem in the future.

Reply:
@jrmajor, would you take a look at the code? Is it a mess? ;-)